tests/baselines/reference/parenthesizedContexualTyping1.types

```diff
@@ -220,11 +220,11 @@ var l = fun(((Math.random() < 0.5 ? ((x => x)) : ((x => undefined)))), ((x => x)
 >x => undefined : (x: number) => any
 >x : number
 >undefined : undefined
->((x => x)) : (x: number) => number
->(x => x) : (x: number) => number
->x => x : (x: number) => number
->x : number
->x : number
+>((x => x)) : (x: any) => any
+>(x => x) : (x: any) => any
+>x => x : (x: any) => any
+>x : any
+>x : any

```

Comment:
It's a subtle interplay between several things. Because we create a new inference type mapper each time we let in an excluded argument, we forget which type parameters were fixed the last time around. Combine that with the fact that once we assign parameter types to a contextually typed function expression, we keep those assignments and don't reassign again. This means that the parameter type assignments that caused type parameter fixing in the previous round don't occur again, and the type parameter now doesn't get fixed, causing us to make more inferences (in this case the `any` that came from the widened `undefined`).

The right thing to do would be to use a single inference mapper for all inferences related to a particular signature and incrementally build up more inferences in that mapper, but remembering which type parameters were already fixed. This would get us more consistent results and also be more efficient. But I'm not sure it needs to be part of this PR, it's really an orthogonal issue.